app/models/r_package.rb

```diff
@@ -24,29 +24,65 @@ def self.database
     call('get_itembank_rdata')
   end
 
-  # Retrieve a hash of attributes defined by the R package
-  # for a given set of answers (e.g. 'EL02' => 1)
+  # Retrieve a hash of attributes defined by the R packag for a given set of answers (e.g. 'EL02' => 1)
   # and domain_ids(e.g. ['POS-PQ'])
   def self.data_for(answers, domain_ids)
     raise 'No domains given' unless domain_ids.present?
 
-    initial_hash = normalized_shadowcat answers: [], domain: domain_ids
+    # TODO: Allow screensmart-r's call_shadowcat function to handle
+    #       multiple domains, which will allow us to simply return
+    #       its return value here, greatly reducing complexity.
+    domain_results = domain_ids.map do |domain_id|
+      data_for_domain(answers, domain_id)
+    end
+
+    { next_question_id: domain_results.find { |dr| !dr[:done] }.try(:[], :next_question_id),
+      done: domain_results.all? { |domain| domain[:done] },
+      domain_results: domain_ids.zip(domain_results).map do |domain_id, domain|
```

Comment:
Dit soort code kan vaak een stuk mooier gemaakt door een deel van de logica naar een class te gooien, waarbinnen je makkelijker kleine subfuncties kan maken. 

maar ik snap dat dit tijdelijke code is?